### ISSUE TITLE
Include authors in intro.md

### ISSUE THREAD

**Issue by rcpeene** [AllenInstitute/openscope_databook#200]
After much investigation, I think the best way to do programmatically do this is to write some python code which extracts this info from the git history, and not go through the github api.

**Comment by rcpeene** [comment]
The git logs for this repo include a few diffferent email addresses for both jerome and I. I'll have to manually code in some resolution of these aliases.

**Comment by rcpeene** [comment]
After a lot of commits, [this script](https://github.com/AllenInstitute/openscope_databook/blob/main/utils/list_authors.py) and [this workflow](https://github.com/AllenInstitute/openscope_databook/actions/workflows/insert_authors.yml) work to update the authors list in intro.md whenever a new author contributes to the repo. The script also includes manually added authors, which are currently @jsiegle and @Ahad-Allen. The current deployment shows four authors.

**Comment by jeromelecoq** [comment]
@rcpeene to write down the issues he faced making this to work

**Comment by rcpeene** [comment]
Okay so some background;

There are some github actions that already exist to log contributors to a repo, such as https://github.com/marketplace/actions/contribute-list and https://github.com/marketplace/actions/contributor_list. Two problems though; These are rather inflexible. We want our contributors list to be pasted at the top of our `intro.md` file to be proudly displayed in our jupyter book as a sort of authors list. Of those two actions, the former presents the contributors as a table including the contributor's photos, and the latter only works on files named README. So these are both almost good contenders for our problem but just a bit too limited.

Additionally, they both go through the Github REST API to get the information. It was @jeromelecoq's wish that we pull our information directly from Git rather than depend on Github to demonstrate our automatic author list. This leads my to the solution; using the `git shortlog` command. Discussed in the next post.

**Comment by rcpeene** [comment]
We wanted a way to determine our repo's contributor's and insert them into our `intro.md` file before each deployment, and we wanted it to be done automatically.

So my solution was to make a python script (linked above) which does this by running some git commands and then editing the `intro.md` file. I then made a github workflow which called the python script and committed changes if there were any. I _could_ maybe have done it in a bash script within a workflow file, but I am not too familiar with bash and it would have been very difficult to read and very extensive.

The problem arises with `git shortlog`. The script calls `git shortlog` as a subprocess call. (There are git libraries for Python, but I didn't manage to find any that had `shortlog` functionality). This ran just fine on my machine, but shortlog failed to produce any results when run within a github workflow. This is because `shortlog` doesn't return results if its not run directly from the command line, _unless_ `git log` is piped into `shortlog` as input.

So after a series of difficulties with python subprocesses and pipes. I managed to get it to work in a github action. Additionally, I encountered the problem of contributor aliases (i.e., my contributions on git show up under "Carter Peene", "Ross Carter Peene", and "rcpeene"). So I also included a blacklist to remove additional usernames and also prevent bots that have committed from showing up as authors.

Finally, I did some tinkering to make sure the workflow only ran before `deploy.yml` gets run, and to only commit to the repo if the authors list was changed.
